holoviews/util/transform.py

```diff
@@ -0,0 +1,293 @@
+import operator
+try:
+    from itertools import zip_longest as zip_longest
+except:
+    from itertools import izip_longest as zip_longest
+
+import numpy as np
+
+from ..core.dimension import Dimension
+from ..core.util import basestring, unique_iterator
+from ..element import Graph
+
+
+def norm_fn(values, min=None, max=None):
+    """
+    min-max normalizes to scale data into 0-1 range.
+
+    Arguments
+    ---------
+    values: np.ndarray
+       Array of values to be binned
+    min: float (optional)
+       Lower bound of normalization range
+    max: float (optional)
+       Upper bound of normalization range
+
+    Returns
+    -------
+    normalized: np.ndarray
+       Array of normalized values
+    """
+    min = np.min(values) if min is None else min
+    max = np.max(values) if max is None else max
+    return (values - min) / (max-min)
+
+
+def bin_fn(values, bins, labels=None):
+    """
+    Bins data into declared bins. By default each bin is labelled
+    with bin center values but an explicit list of bin labels may be
+    defined.
+
+    Arguments
+    ---------
+    values: np.ndarray
+       Array of values to be binned
+    bins: np.ndarray or list
+       Bin edges to place bins into
+    labels: np.ndarray or list (optional)
+       Labels for bins should have length N-1 compared to bins
+
+    Returns
+    -------
+    binned: np.ndarray
+       Array of binned values
+    """
+    bins = np.asarray(bins)
+    if labels is None:
+        labels = (bins[:-1] + np.diff(bins)/2.)
+    else:
+        labels = np.asarray(labels)
+    dtype = 'float' if labels.dtype.kind == 'f' else 'O'
+    binned = np.full_like(values, (np.nan if dtype == 'f' else None), dtype=dtype)
+    for lower, upper, label in zip(bins[:-1], bins[1:], labels):
+        condition = (values > lower) & (values <= upper)
+        binned[np.where(condition)[0]] = label
+    return binned
+
+
+def cat_fn(values, categories, empty=None):
```

Comment:
Here I'd would have suggested just `cat` instead of `cat_fn`, and I do think that would be better, but I personally cannot think of `cat` on a computer as anything other than the UNIX command by that name, and so it expands to "concatenate" in my mind, not "categorize" or "categorical".  I'd vote for calling this function and the corresponding method `categorical`, because they turn what they are given into a categorical value, right?  I don't think it's such a common operation that it needs to shave off the last few characters, when that makes it confusing as in this case...

Reply:
I think I prefer `categorize`, but it is a bit of shame since ``dim('value').categorize(['o', 'x', '+'])`` is significantly longer than `dim('value').cat(['o', 'x', '+'])`. You're definitely right that `cat` usually refers to concatenate though, that's true even in pandas.